util/lua/impacted_resource.go

```diff
@@ -1,50 +1,60 @@
 package lua
 
 import (
+	"encoding/json"
 	"fmt"
 
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
-// This struct represents a wrapper, that is returned from Lua custom action script, around the unstructured k8s resource + a k8s operation
-// that will need to be performed on this returned resource.
-// Currently only "create" and "patch" operations are supported for custom actions.
-// This replaces the traditional architecture of "Lua action returns the source resource for ArgoCD to patch".
-// This enables ArgoCD to create NEW resources upon custom action.
-// Note that the Lua code in the custom action is coupled to this type, since Lua json output is then unmarshalled to this struct.
-// Avoided using iota, since need the mapping of the string value the end users will write in Lua code ("create" and "patch").
-// TODO: maybe there is a nicer general way to marshal and unmarshal, instead of explicit iteration over the enum values.
+// K8SOperation represents the operation type for k8s resources.
 type K8SOperation string
 
+// Currently, only "create" and "patch" operations are supported.
```

Comment:
```suggestion
// Currently, only "create" and "patch" operations are supported.
```
just wondering why aren't the other operations supported?

Reply:
This PR is purely a refactor. The goal was to clean up the (un)marshal logic and make it more maintainable, not to change or expand the supported functionality. I’ve kept the logic and behavior exactly the same — just improved how it’s implemented internally.
But i think that only "create" and "patch" are supported because they cover the main use cases for Lua-based custom actions in Argo CD — creating new resources or modifying existing ones in a safe and controlled way. Other operations like "delete" or "replace" could have broader or destructive effects (maybe the reason which is why they're not part of the current design?).
Maybe the design could be extended later if there's a good reason to support more operations?